include/lbann/layers/transform/mean_squared_error.hpp

```diff
@@ -0,0 +1,133 @@
+                 cudnn::cudnn_manager *cudnn = nullptr)
+    : transform_layer(comm) {
+
+    // 2 parents expected: ground truth/label and predictions
+    m_expected_num_parent_layers = 2;
+
+  #ifdef LBANN_HAS_CUDNN
+    // Activate GPU if needed
+    if (cudnn != nullptr) {
+      this->m_cudnn = cudnn;
+    }
+  #endif // LBANN_HAS_CUDNN
+
+  }
+
+  mean_squared_error_layer* copy() const override { return new mean_squared_error_layer(*this); }
+  std::string get_type() const override { return "Binary cross entropy layer"; }
+  data_layout get_data_layout() const override { return T_layout; }
+  El::Device get_device_allocation() const override { return Dev; }
+
+  /** Returns description of ctor params */
+  std::string get_description() const override {
+    std::stringstream s;
+    s << " MSE layer ; parents: ";
+    for (size_t i=0; i<this->m_parent_layers.size(); i++) {
+      s << this->m_parent_layers[i]->get_name() << " " << this->m_parent_layers[i]->get_type() << " ";
+    }
+    s << " dataLayout: " << this->get_data_layout_string(get_data_layout());
+    return s.str();
+  }
+
+  protected:
+void fp_compute() {
+  const auto& predictions = get_prev_activations(0);
+  const auto& ground_truth = get_prev_activations(1);
+  auto& output_local = get_activations().Matrix();
+  // Local matrices
+  const Mat& predictions_local = predictions.LockedMatrix();
+  const Mat& ground_truth_local = ground_truth.LockedMatrix();
+  
+  // Matrix parameters
+  const int local_height = predictions_local.Height();
+  const int local_width = predictions_local.Width();
+
+  // Compute squared sum of errors
+  #pragma omp parallel for collapse(2)
+  for (int col = 0; col < local_width; ++col) {
+    for (int row = 0; row < local_height; ++row) {
+      const DataType true_val = ground_truth_local(row, col);
+      const DataType pred_val = predictions_local(row, col);
+      const DataType error = true_val - pred_val;
+      output_local(row,col) += error * error;
+    }
+  }
+
+}
+
+void bp_compute() {
+
+  const auto& predictions = get_prev_activations(0);
+  const auto& ground_truth = get_prev_activations(1);
+  auto& gradient = get_error_signals();
+  // Local matrices
+  const Mat& predictions_local = predictions.LockedMatrix();
+  const Mat& ground_truth_local = ground_truth.LockedMatrix();
+  auto& gradient_local = gradient.Matrix();
+
+  // Matrix parameters
+  const int height = gradient.Height();
+  const El::Int local_height = gradient_local.Height();
+  const El::Int local_width = gradient_local.Width();
+
+  // Compute gradient
+  const DataType scale = DataType(2) / height; 
+  #pragma omp parallel for collapse(2)
+  for (El::Int col = 0; col < local_width; ++col) {
+    for (El::Int row = 0; row < local_height; ++row) {
+      const DataType true_val = ground_truth_local(row, col);
+      const DataType pred_val = predictions_local(row, col);
+      gradient_local(row, col) = (pred_val - true_val) * scale;
```

Comment:
We need to multiply this by the previous error signal:
dL/d(pred) = 2 * (pred - label) / num_neurons * sum(dL/d(output))

We should also compute the gradient w.r.t. the ground truth:
dL/d(label) = 2 * (label - pred) / num_neurons * sum(dL/d(output))